mps_youtube/listview.py

```diff
@@ -0,0 +1,237 @@
+            even if the item does not have one.
+        """
+        return 0
+
+
+class ListUser(ListViewItem):
+    """ Describes a user
+    """
+    # pylint: disable=unused-argument
+    def id(self, length=0):
+        """ Returns YTID """
+        return self.data.get("id").get("channelId")
+
+    def name(self, length=10):
+        """ Returns channel name """
+        return util.uea_pad(length, self.data.get("snippet").get("title"))
+
+    def description(self, length=10):
+        """ Channel description"""
+        return util.uea_pad(length, self.data.get("snippet").get("description"))
+
+    def kind(self, length=10):
+        """ Returns the youtube datatype
+            Example: youtube#channel, youtube#video
+        """
+        return self.data.get("id").get("kind")
+
+    def ret(self):
+        """ Used in the ListView play function """
+        return (self.data.get("snippet").get("title"), self.id(), "")
+
+    @staticmethod
+    def return_field():
+        """ Determines which function will be called on selected items """
+        return "ret"
+
+
+class ListView(content.Content):
+    """ Content Agnostic Numbered List
+
+        This class, using ListViewItems as abstractions you can
+        give it a list of data and which columns to show and it will
+        show it.
+
+        Todo:
+            Currently we rely on the commands/play code to send information
+            about which elements are being picked.
+
+        Attributes:
+            func        The function that will be run on the selected items
+            objects     List of objects(or a ContentQuery object)
+            columns     A list of Hashes containing information about which
+                        columns to show
+            page        Current Page
+
+        Column format:
+            {"name": "idx", "size": 3, "heading": "Num"}
+            name:    The method name that will be called from the ListViewItem
+            size:    How much size is allocated to the columns,
+                     see ListView.content for more information about
+                     the dynamic options
+            heading: The text shown in the header
+
+            "idx" is generated in the content function, not by the ListViewItem
+    """
+    func = None
+    objects = None
+    columns = None
+    page = 0
+
+    def __init__(self, columns, objects, function_call=None):
+        """ """
+        self.func = function_call
+        self.objects = objects
+        self.columns = columns
+        self.object_type = None
+
+        # Ensure single type of object
+        types = len(set([obj.__class__ for obj in objects]))
+        if types == 0:
+            raise BaseException("No objects in list")
+        if types > 1:
+            raise BaseException("More than one kind of objects in list")
+
+        self.object_type = [obj.__class__ for obj in objects][0]
+        del types
+
+    def pages(self):
```

Comment:
This function is just the equivalent of `max(1, len(self.objects) / self.views_per_page())`, right?